mesonbuild/templates/samplefactory.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from mesonbuild.templates.fortrantemplates import FortranProject
+from mesonbuild.templates.objcpptemplates import ObjCppProject
+from mesonbuild.templates.dlangtemplates import DlangProject
+from mesonbuild.templates.rusttemplates import RustProject
+from mesonbuild.templates.javatemplates import JavaProject
+from mesonbuild.templates.cudatemplates import CudaProject
+from mesonbuild.templates.objctemplates import ObjCProject
+from mesonbuild.templates.cpptemplates import CppProject
+from mesonbuild.templates.cstemplates import CSharpProject
+from mesonbuild.templates.ctemplates import CProject
+
+
+class SampleFactory(object):
+    @staticmethod
+    def sameple_generator(options):
+        if options.language == 'c':
+            return CProject(options)
+        elif options.language == 'cpp':
+            return CppProject(options)
+        elif options.language == 'cs':
+            return CSharpProject(options)
+        elif options.language == 'objc':
+            return ObjCProject(options)
+        elif options.language == 'objcpp':
+            return ObjCppProject(options)
+        elif options.language == 'java':
+            return JavaProject(options)
+        elif options.language == 'd':
+            return DlangProject(options)
+        elif options.language == 'rust':
+            return RustProject(options)
+        elif options.language == 'cuda':
+            return CudaProject(options)
+        elif options.language == 'fortran':
+            return FortranProject(options)
```

Comment:
I'd recommend something like this instead
```python
return {
    'c': CProject,
    'cpp': CppProject,
    .....
}[options.language](options)
```

Reply:
Like the suggestion.  I never have seen `[options.language](options)` syntax before.  Is it compatible with Python 3.5?

Reply:
It's not any special syntax. You're just instantiating a dict object `{...}`, then indexing it `[...]`, which returns one of the class objects, which you then call/instantiate `(...)`. Ought to be working for as long as Python had dict literals. You can rewrite it like that for clarity:

```python
classes = {
    'c': CProject,
    'cpp': CppProject,
    .....
}
proj_class = classes[options.language]
proj = proj_class(options)
return proj
```

Reply:
I will see if I can get that to work otherwise thanks for the suggestion.